src/js/angular/ttyg/directives/chat-view.directive.js

```diff
@@ -0,0 +1,179 @@
+import {ChatQuestion} from "../../models/ttyg/chat-question";
+import {TTYGEventName} from "../services/ttyg-context.service";
+import {chatQuestionToChatMessageMapper} from "../services/chat-message.mapper";
+import {cloneDeep} from "lodash";
+
+const modules = [];
+
+angular
+    .module('graphdb.framework.ttyg.directives.chat-view', modules)
+    .directive('chatView', ChartViewComponent);
+
+ChartViewComponent.$inject = ['toastr', '$translate', 'TTYGContextService', 'TTYGService'];
+
+/**
+ * @ngdoc directive
+ * @name graphdb.framework.ttyg.directives.chat-view:chatView
+ * @restrict E
+ * @description
+ *
+ * This directive represents a chat view component that allows users to interact with a chat. It provides functionality for asking questions,
+ * regenerating chat content, and other interactive features.
+ *
+ * It encapsulates the logic and presentation for displaying and managing chat-related interactions.
+ *
+ * @example
+ * <chat-view></chat-view>
+ */
+function ChartViewComponent(toastr, $translate, TTYGContextService, TTYGService) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/ttyg/templates/chat-view.html',
+        scope: {},
+        link: ($scope, element, attrs) => {
+
+            // =========================
+            // Public variables
+            // =========================
+
+            /**
+             * @type {ChatModel}
+             */
+            $scope.selectedChat = undefined;
+
+            /**
+             * @type {AgentModel}
+             */
+            $scope.selectedAgent = undefined;
+
+            /**
+             * @type {ChatQuestion}
+             */
+            $scope.chatQuestion = undefined;
+
+            // =========================
+            // Private variables
+            // =========================
+
+            // =========================
+            // Public functions
+            // =========================
+
+            /**
+             * Handles the ask question action.
+             */
+            $scope.ask = () => {
+                if (!$scope.chatQuestion.conversationId) {
+                    createNewChat();
+                } else {
+                    askQuestion();
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+
+            const setupNewChatQuestion = () => {
+                $scope.chatQuestion = getEmptyChatQuestion();
+            };
+
+            const createNewChat = () => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT, $scope.chatQuestion);
+            };
+
+            const askQuestion = () => {
+                if ($scope.selectedChat) {
+                    $scope.selectedChat.messages.push(chatQuestionToChatMessageMapper($scope.chatQuestion));
```

Comment:
Create an `appendMessage` method in the chat model. Don't leak the model internals in the code.

Reply:
Done